### ISSUE TITLE
3D axis limits on Linux not handled properly

### ISSUE THREAD

**Issue by DanielMurphy22** [mantidproject/mantid#30014]
Found when creating #29926  

### Expected behavior
Double-click menus to edit axis limits on 3D plots work smoothly on Linux
### Actual behavior
Works fine on Windows and Mac, but on Linux (and IDAaaS) it doesn't work correctly
as the matplotlib version is on 2.2.3. (MAc and Windows are on 3.1.3 and 3.1.2 respectively)
### Steps to reproduce the behavior
Run: 
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

ws = CreateSampleWorkspace(Function = 'Exp Decay')
fig, ax = plt.subplots(subplot_kw={'projection':'mantid3d'})
ax.plot_surface(ws)
plt.show()
```
then double-click on the z axis values and enter `Max = 0.2`

error message: `'MantidAxes3DSubplot' object has no attribute 'get_zaxis'`
### Platforms affected
Linux / IDAaaS Workbench 5.1.1 and nightly

**Comment by DavidFair** [comment]
Unstaling, as this was found in #33187

**Comment by martyngigg** [comment]
Fixed by #33259
